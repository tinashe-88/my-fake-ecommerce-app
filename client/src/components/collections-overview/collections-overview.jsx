import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import CollectionPreview from '../collection-preview/collection-preview'

import { selectCollectionForPreview } from '../../redux/shop/shop.selector'

import { CollectionsOverviewContainer } from './collections-overview.styles'

export const CollectionsOverview = ({ collections }) => (
  <CollectionsOverviewContainer>
    {collections.map(({ id, ...otherCollectionProps }) => (
      <CollectionPreview key={id} {...otherCollectionProps} />
    ))}
  </CollectionsOverviewContainer>
)

const mapStateToProps = createStructuredSelector({
  collections: selectCollectionForPreview
})

export default connect(mapStateToProps)(CollectionsOverview)