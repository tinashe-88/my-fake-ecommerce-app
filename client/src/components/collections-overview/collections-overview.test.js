import React from 'react'
import { shallow } from 'enzyme'

import { CollectionsOverview } from './collections-overview'

it('Should render the CollectionsOverview component', () => {
  expect(shallow(<CollectionsOverview collections={[]}/>)).toMatchSnapshot()
})