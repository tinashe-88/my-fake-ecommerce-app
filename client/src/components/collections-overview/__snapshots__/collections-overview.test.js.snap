// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the CollectionsOverview component 1`] = `<styled.div />`;
